src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,78 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a task, which can be a todo, deadline or event.
+ */
+public abstract class Task {
+    private final String description;
+    private boolean isDone;
+
+    /**
+     * The task constructor.
+     *
+     * @param description The description of the task.
+     * @param isDone The boolean keeping track of whether the task is done.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public abstract boolean isDue(LocalDate date);
+
+    /**
+     * Returns a boolean indicating if the task contains the given keyword.
+     *
+     * @param keyword A keyword.
+     * @return true if the task contains the given keyword.
+     */
+    public boolean hasKeyword(String keyword) {
+        return this.description.contains(keyword);
+    }
+
+    /**
+     * Returns a status icon corresponding to whether the task is done.
+     *
+     * @return a tick if the task is done, and a cross otherwise.
+     */
+    public String getStatusIcon() {
+        return this.isDone ? "\u2713" : "\u2718";
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Returns a string representation of the task for saving into a file.
+     *
+     * @return a string representation of the task.
+     */
+    public String toSaveData() {
+        return (this.isDone ? 1 : 0) + " | " + this.description;
```

Comment:
Might want to use something "rarer" for separator, to avoid error parsing tasks with "|" inside.

Reply:
Actually, I believe my parser accounts for that since the description is last and I only split the line using the first 2 instances of "|".